### ISSUE TITLE
Suspend to RAM has no way to wake up

### ISSUE THREAD

**Issue by nuclearsandwich** [nuclearsandwich/surface3-archlinux#6]
Suspending to RAM via systemd apparently has no way to wake up. The power button is not triggering the computer to wake up.

I guess I haven't 100% confirmed that the computer isn't resuming and the backlight is staying off. Next time I try I will attempt to reboot blind after suspending from a root terminal.

**Comment by paulhandy** [comment]
There does seem to be a* way to wake up, and that is fn keys on a usb keyboard that was never unplugged after suspending. Hardly useful, but it is something.

One thing that bothers me, is that it seems to suspend when I close the type cover 3. It doesn't resume when I open the type cover 3, however, which seems a bit backwards to me.

**Comment by shvr** [comment]
Using the "OneNote" button on the pen wakes the SP3 for me, but it has it's flaws. Resuming pretty much always disables the touchscreen and more often then not / get mounted read only. GDM will also sometimes refuse to restart. It's pretty buggy, but it "works".

For the record, I'm using kernel 3.19-rc2, GNOME 3.14, and suspending via the GNOME power menu.

**Comment by altercation** [comment]
@shvr You may have just changed my life. Embarrassed to say that I never used the pen in Linux (only in windows dual booted). I just paired it and everything works great other than the power hit due to BT. It may not be perfect but it's better than just powering off all the time. THANK YOU.

I'm testing resume functionality now but basically as long as I drop wifi prior to suspend it seems pretty stable. I'm running i3 fwiw.

**Comment by shvr** [comment]
No problem. Was just something I thought I'd try after reading this thread and it happened to work. Hopefully though the button issues get worked out soon so this workaround isn't needed.

**Comment by altercation** [comment]
I'm guessing that I could assign any bt device to wakeup via udev rules, though I haven't tested this and other than a bluetooth mouse I don't have anything I think would be as convenient as the pen.

**Comment by paulhandy** [comment]
The "oneNote" button does nothing for me when suspended, with 3.17.3-pf kernel patched with type cover and camera patches (am i missing any other patches?). Did you make any special udev rules for it to work?

**Comment by shvr** [comment]
Nothing special was needed. Just paired the pen and went. Did you download and install the Marvell wifi / bluetooth from their git repository?

**Comment by paulhandy** [comment]
Yes, I installed that software. Perhaps something is different in the newer kernel?

**Comment by shvr** [comment]
I downgraded my kernel to 3.17.6 (the one currently in Arch's Core repo) and tried this again. I was able to successfully wake from sleep using the OneNote button on the pen, so I don't think the kernel is to blame. You may try checking if you have a setting enables that also puts bluetooth to sleep when suspending as well as verifying that the pen is actually paired and can connect.

**Comment by shvr** [comment]
To clarify, I used the stock Arch kernel and applied no patches.

**Comment by altercation** [comment]
To aid with troubleshooting, I'm using blueman as a bt management tray icon/ui and there is a handy data in/out display on it that clearly increments with each press of my pen button (I'm sure you could achieve the same monitoring on the command line as well).

**Comment by paulhandy** [comment]
Does anyone know how to debug wether bluetooth is set to sleep when the computer sleeps?

**Comment by altercation** [comment]
New patches just released by the team working on the SP3 power button seem to have done the trick. Button is working for suspend/resume here.

https://bugzilla.kernel.org/show_bug.cgi?id=84651

**Comment by colorprint** [comment]
Yes, suspend work fine, but hibernate (resume after hibernate) seems to not works.
Also I see strange issue with keyboard cover with these pathes - if I boot the Surface with keyboard cover attached - it hangs when I try to login in gnome. If I boot it with detached keyboard cover and attach it at login screen - it works fine.

**Comment by paulhandy** [comment]
Dang. Right after my company took it back, too. Oh well.

On Fri, Mar 20, 2015 at 10:00 AM, Anton Anikin <notifications@github.com>
wrote:

> Yes, suspend work fine, but hibernate (resume after hibernate) seems to
> not works.
> Also I see strange issue with keyboard cover with these pathes - if I boot
> the Surface with keyboard cover attached - it hangs when I try to login in
> gnome. If I boot it with detached keyboard cover and attach it at login
> screen - it works fine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nuclearsandwich/surface3-archlinux/issues/6#issuecomment-84055683>
> .
>



-- 

   ______________________________________________________________________
  | .     .               .   .                   .          .           |
  |              . .                     .      ___,_   _         .   .  |
  | .                       .      .          [:t_:::;t"t"+        .     |
  |      .                     .            . `=_ "`[ j.:\=\             |
  |             .      .              .        _,:-.| -"_:\=\  .         |
  |    .           .          .           _,-=":.:%.."+"+|:\=\        .  |
  |          .                   _ _____,:,,;,==.==+nnnpppppppt          |
  |                           _.;-^-._-:._::.'';nn;::m;:%%%%%%%\   .     |
  |  .       .              .;-'_::-:_"--;_:. ((888:(@) ,,;::^%%%,       |
  |                      __='::_:"`::::::::"-;_`YPP::; (d8B((@b."%\     .|
  |      ,------..    __,-:-:::::::::`::`::::::"--;_(@' 88P':^" ;nn:,    |
  |   ,-":%%%%::==.  ;-':::::`%%%\::---:::-:_::::::_"-;_.::((@,(88J::\   |
  |  /:::__ ::%::== """"""""""""""`------`.__.-:::::;___;;::`^__;;;:..7  |
  | /::.'  `.:%%=:=`-=,     . i                   .       """"           |
  |Y:::f    j :%%%%:::=::    ,^.    .        |-|  . .                    |
  |l   `.__+ :::%%%%:::_;[                        |o|                    |
  ||^~'-------------""~:^|                       _` ` _  .. __,,,,+++O#@@#
  |! ::::::::::%%%%==:{                       __j [,,j [#O|||O#@@#O++:|@@#
  | \ `::====: ==== :='            .__,,,++::::.j "  " [%+++::|@##O+::+O##
  |  \:== :: == :=='    __,,,+++|O|. +++..   :::j_[nnj_[_++:+%%_%%|+%|%+O#
  |   "-. =_:::: },+|O##|+::+|:++:::..    ::: .:+%%%%%%j [%O%%j [:+++|++|O
  |   _,,`-------' .+#O#+:||%+        :: .. .:++|O###O%j `'  `' [:::::::++
  |.+:..:++|++||||+.O.::++:|::..:...:...:++++|||+O##||%j [%..%j [+::LS:+%|

**Comment by lateau** [comment]
On 4.1.6 I have two suspend issues. Suspend does not work well with systemctl command, it causes freezing display and never go to asleep. By closing a typecover, my fully charged battery discharged in about 7 hours and it means suspend doesn't work with that.

I installed with 2015-09 Arch Linux Image and no patches are applied. Is there anything that I should do some additional tasks?
